api/controllers/PostController.js

```diff
@@ -392,6 +400,36 @@ module.exports = {
       }
     })
     .then(() => res.ok({}), res.serverError)
-  }
+  },
+
+  checkFreshnessForCommunity: function (req, res) {
+    var foundPosts
+    var resolve = value => foundPosts = value
```

Comment:
this smells to me. I feel like there's a way to do it that's more Promisey, but I can't quite work it out. Whatever it is would likely require another refactor for findPosts. Any thoughts?